src/main/java/uk/gov/hmcts/darts/arm/mapper/impl/TranscriptionArchiveRecordMapperImpl.java

```diff
@@ -60,14 +130,216 @@ private UploadNewFileRecordMetadata createUploadNewFileRecordMetadata(Transcript
+        }
+        if (transcriptionRecordProperties.containsKey(BF_018_KEY)) {
+            metadata.setBf018(mapToString(transcriptionRecordProperties.getProperty(BF_018_KEY), transcriptionDocument));
+        }
+        if (transcriptionRecordProperties.containsKey(BF_019_KEY)) {
+            metadata.setBf019(mapToString(transcriptionRecordProperties.getProperty(BF_019_KEY), transcriptionDocument));
+        }
+        if (transcriptionRecordProperties.containsKey(BF_020_KEY)) {
+            metadata.setBf020(mapToString(transcriptionRecordProperties.getProperty(BF_020_KEY), transcriptionDocument));
+        }
+        return metadata;
+    }
+
+    private String mapToString(String key, TranscriptionDocumentEntity transcriptionDocument) {
+        return switch (key) {
+            case OBJECT_TYPE_KEY -> ArchiveRecordType.TRANSCRIPTION_ARCHIVE_TYPE.getArchiveTypeDescription();
+            case FILE_TYPE_KEY -> transcriptionDocument.getFileType();
+            case HEARING_DATE_KEY -> getHearingDate(transcriptionDocument);
+            case CHECKSUM_KEY -> transcriptionDocument.getChecksum();
+            case TRANSCRIPT_REQUEST_KEY -> getTranscriptionRequest(transcriptionDocument);
+            case TRANSCRIPT_TYPE_KEY -> getTranscriptionType(transcriptionDocument);
+            case TRANSCRIPT_URGENCY_KEY -> getTranscriptionUrgency(transcriptionDocument);
+            case COMMENTS_KEY -> getTranscriptionComments(transcriptionDocument);
+            case CREATED_DATE_TIME_KEY -> getCreatedDateTime(transcriptionDocument);
+            case UPLOADED_BY_KEY -> getUploadedBy(transcriptionDocument);
+            case START_DATE_TIME_KEY -> getStartDateTime(transcriptionDocument);
+            case END_DATE_TIME_KEY -> getEndDateTime(transcriptionDocument);
+            case COURTHOUSE_KEY -> getCourthouse(transcriptionDocument);
+            case COURTROOM_KEY -> getCourtroom(transcriptionDocument);
+            default -> null;
+        };
+    }
+
+    private String getEndDateTime(TranscriptionDocumentEntity transcriptionDocument) {
+        String endDateTime = null;
+        if (nonNull(transcriptionDocument.getTranscription()) && nonNull(transcriptionDocument.getTranscription().getEndTime())) {
+            endDateTime = transcriptionDocument.getTranscription().getEndTime().format(dateTimeFormatter);
+        }
+        return endDateTime;
+    }
+
+    private String getStartDateTime(TranscriptionDocumentEntity transcriptionDocument) {
+        String startDateTime = null;
+        if (nonNull(transcriptionDocument.getTranscription()) && nonNull(transcriptionDocument.getTranscription().getStartTime())) {
+            startDateTime = transcriptionDocument.getTranscription().getStartTime().format(dateTimeFormatter);
+        }
+        return startDateTime;
+    }
+
+    private static String getUploadedBy(TranscriptionDocumentEntity transcriptionDocument) {
```

Comment:
should this be called getUploadedByFullName

Reply:
Actually I'm not quite sure what data we should be sending as its a data privacy issue

Reply:
I have changed it to ID for the moment so we are not sending personal info